sklearn/discriminant_analysis.py

```diff
@@ -475,7 +477,7 @@ def _solve_svd(self, X, y):
         # Centers are living in a space with n_classes-1 dim (maximum)
         # Use SVD to find projection in the space spanned by the
         # (n_classes) centers
-        _, S, Vt = linalg.svd(X, full_matrices=0)
+        _, S, Vt = linalg.svd(X, full_matrices=0, check_finite=False)
```

Comment:
If this https://github.com/scikit-learn/scikit-learn/issues/18837#issuecomment-734362140 is valid the input may be non finite

Reply:
I didn't thought of that, and I think you are right. I'm waiting for confirmation but it seems this whole PR should not be made. Thanks for catching this !

Reply:
This should be OK as well.